### PR TITLE
Add Win32 API names dictionary

### PR DESCRIPTION
**Pull request by LuizZak**
~~(in WIP while I investigate the generator not collecting enumerators behind typedefs correctly.)~~

When working with Win32 APIs, a lot of the commonly used nomenclature and identifiers can be triggered by the spell checker, like `LPCWSTR`, `HRESULT`, `wcsnicmp`, `uuidof`, etc., so I decided to tackle that with a ready-to-use, comprehensive dictionary that can be easily generated by code.

I created a python generator that sweeps an installed Windows SDK and extracts publicly-available APIs and constants to generate a (quite hefty, actually) dictionary that I hope will be useful for people doing low-level Win32 development.

I'm unsure if this is fit for this repository or if I'm better off creating a separate repo myself, so I'm opening a PR to check if it's an ok addition.

### CODE REVIEW

File: dictionaries/swift/CHANGELOG.md [hunk 1]
```diff
@@ -1,16 +1,3 @@
-# Change Log
```
Comment (Jason3S) [inline dictionaries/swift/CHANGELOG.md:1]: I think this file was changed by accident.

File: dictionaries/swift/package.json [hunk 2]
```diff
@@ -1,7 +1,8 @@
 {
```
Comment (Jason3S) [inline dictionaries/swift/package.json:1]: This one too, was changed by accident.

File: dictionaries/win32/samples/stdafx.h [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
+// ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
+// IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
+// PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.
+//
+//*********************************************************
+
+// stdafx.h : include file for standard system include files,
```
Comment (Jason3S) [inline dictionaries/win32/samples/stdafx.h:12]: `stdafx` really should be part of the dictionary 😊

File: dictionaries/win32/src/generate.py [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+def expand(line):
+    # Do camelCase/PascalCase expansion
+    matches = re.findall(r'.+?(?:(?<=[a-z])(?=[A-Z])|(?<=[A-Z])(?=[A-Z][a-z])|$)', line)
+    # Split snake_case strings
+    matches = flatten([term.split("_") for term in matches])
+    # Split around numbers
+    matches = flatten([re.split(r'[0-9]+', term) for term in matches])
+
+    return [term.lower() for term in matches if len(term) > 3 and not term.isnumeric()]
```
Comment (Jason3S) [inline dictionaries/win32/src/generate.py:49]: It is better to keep the case. The spell checker is case sensitive.

Comment (Jason3S) [reply dictionaries/win32/src/generate.py:49]: See https://github.com/streetsidesoftware/cspell-dicts/issues/589#issuecomment-945059578 for a discussion.

**Comment by Jason3S** [comment]
Thanks for doing this.

File: dictionaries/win32/samples/stdafx.h (hunk 3 above)
Comment (LuizZak) [reply dictionaries/win32/samples/stdafx.h:12]: Added!

**Comment by Jason3S** [comment]
@LuizZak,

I think a Win32 dictionary is useful for other people as well.

**Pull request merged**